### ISSUE TITLE
Convert modals that currently aren't using the Preact modal to the Preact modal

### ISSUE THREAD

**Issue by nickytonline** [forem/forem#16328]
## Description

There's work being done in #16310 for modals and one thing that came up was that there are still some modals in the codebase that aren't using the Preact modal component.

These non-Preact modals should be converted to the Preact modal component, not just for keeping things uniform, but also because the Preact modal component has a tonne of accessibility baked in to it.

_Originally posted by @nickytonline in https://github.com/forem/forem/pull/16310#r793649624_

**Comment by aitchiss** [comment]
FYI - you can find these "legacy" modals by searching for use of the `crayons-modal` class name. A quick search today shows deprecated non-Preact modals in:

- `app/javascript/admin/adminModal.js`
- `app/views/liquids/_user_subscription.html.erb`
- `app/views/users/show.html.erb`

There are also usages in the onboarding flow, but this might be best looked at separately as they're a very different design/usage than other app modals (do they even need to be modals at all?).
